components/camel-cdi/src/main/java/org/apache/camel/cdi/transaction/TransactionalJavaEETransactionPolicy.java

```diff
@@ -0,0 +1,121 @@
+package org.apache.camel.cdi.transaction;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.RollbackException;
+import javax.transaction.Status;
+import javax.transaction.SystemException;
+import javax.transaction.Transaction;
+
+import org.apache.camel.CamelException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper methods for transaction handling
+ */
+public abstract class TransactionalJavaEETransactionPolicy extends JavaEETransactionPolicy {
```

Comment:
I wonder to what extent that'd make sense to have that class merged with `JavaEETransactionPolicy` and have it renamed `JtaTransactionPolicy`. Indeed, I would not expect an non transactional JTA transaction policy.